### PR TITLE
 [PR for comments] Switch from attoparsec to cats-parse

### PR DESCRIPTION
**Pull request by stephenjudkins**

### CODE REVIEW

File: core/src/main/scala/sqlitis/Parser.scala [hunk 1]
```diff
@@ -46,83 +71,92 @@ object Parser {
...
+    )
 
-  private def kw(s: String) = token(stringCI(s))
+  private def kw(s: String): Parser1[String] = token(stringCI(s)).backtrack
 
-  private def optKw[A](k: String, parser: Parser[A]): Parser[Option[A]] =
-    opt(kw(k)).flatMap {
+  private def optKw[A](k: String, parser: P[A]): P[Option[A]] =
+    kw(k).?.flatMap {
       case Some(_) => parser.map(Some(_))
```
Comment (johnynek) [inline core/src/main/scala/sqlitis/Parser.scala:88]: I don't think you need flatMap here... isn't this:

```scala
(kw(k) *> parser).? <* maybeWhitespace
```

File: core/src/main/scala/sqlitis/Parser.scala [hunk 2]
```diff
@@ -46,83 +71,92 @@ object Parser {
...
-    case (((((isDistinct, fields), from), where), orderBy), limit) =>
-      Select(
-        isDistinct = isDistinct,
-        fields = fields,
-        from = from,
-        where = where,
-        orderBy = orderBy,
-        limit = limit
-      )
+  ).map { case (((((isDistinct, fields), from), where), orderBy), limit) =>
```
Comment (johnynek) [inline core/src/main/scala/sqlitis/Parser.scala:143]: I'd use cats syntax here and do, `(p1, p2, p3, ...).mapN { (a1, a2, ...`

and not have to deal with the nesting personally.  But this is only a syntactic change.

**Comment by stephenjudkins** [comment]
@johnynek thanks a bunch! applied all of your suggestions and I think it looks a lot nicer https://github.com/stephenjudkins/sqlitis/commit/d41f4bdd5036ff8b680ba4b298ef094b955aa41d

File: core/src/main/scala/sqlitis/Parser.scala [hunk 3]
```diff
@@ -133,14 +131,14 @@ object Parser {
...
-  implicit lazy val select: P[Select[Unit]] = (
-    kw("SELECT") *> (kw("DISTINCT")).?.map(_.isDefined) ~
-      repSep(field, 0, comma) ~
-      optKw("FROM", repSep(from, 1, comma)).map(_.getOrElse(Nil)) ~
-      optKw("WHERE", expr <* maybeWhitespace) ~
-      optKw("ORDER BY", orderBy) ~
-      optKw("LIMIT", token(int))
-  ).map { case (((((isDistinct, fields), from), where), orderBy), limit) =>
+  implicit val select: P[Select[Unit]] = (
+    kw("SELECT") *> (kw("DISTINCT")).?.map(_.isDefined),
```
Comment (johnynek) [inline core/src/main/scala/sqlitis/Parser.scala:135]: I think you wanted `kw("SELECT")` outside of the `(...` which would then make the resulting parser a Parser1, which can be useful sometimes...

**Pull request closed without merging**